src/test/java/org/csanchez/jenkins/plugins/kubernetes/pipeline/KubernetesPipelineTest.java

```diff
@@ -204,6 +204,28 @@ public void runWithOverriddenNamespace() throws Exception {
         r.assertLogContains(overriddenNamespace, b);
     }
 
+    @Test
+    public void runWithOverriddenNamespace2() throws Exception {
```

Comment:
`runWithOverriddenNamespace` is meant to verify that if a namespace is not explicitly specified it will be picked up from the `NamespaceAction`.

`runWithOverriddenNamespace2` is meant to verift that if a namepsace IS explictly specified its the one that will be used.